pkg/parse/source.go

```diff
@@ -96,6 +96,9 @@ func (o *files) readConfigFiles(state *sourceState) status.Error {
 	if err != nil {
 		return status.PathWrapError(errors.Wrap(err, "listing files in the configs directory"), syncDir.OSPath())
 	}
+	if state.commit != commit {
+		return status.ObsoleteFileErrorf("Commit/image digest/chart version has changed when listing the files, files not updated")
```

Comment:
The error message will be visible to the user. Suggest making it more user-friendly.

Reply:
Aligned with hydration controller.